DSLink.Tests/NodeTests.cs

```diff
@@ -0,0 +1,239 @@
+﻿using DSLink.Connection;
+using DSLink.Container;
+using DSLink.Nodes;
+using DSLink.Respond;
+using Moq;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using System;
+using System.Threading.Tasks;
+
+namespace DSLink.Tests
+{
+    [TestFixture]
+    public class NodeTests
+    {
+        private Mock<AbstractContainer> _mockContainer;
+        private Mock<Connector> _mockConnector;
+        private Mock<Responder> _mockResponder;
+        private Mock<SubscriptionManager> _mockSubManager;
+        private Node _superRootNode;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockContainer = new Mock<AbstractContainer>();
+            _mockConnector = new Mock<Connector>(_mockContainer.Object);
+            _mockResponder = new Mock<Responder>();
+            _mockSubManager = new Mock<SubscriptionManager>(_mockContainer.Object);
+            _superRootNode = new Node("", null, _mockContainer.Object);
+
+            _mockContainer.SetupGet(c => c.Connector).Returns(_mockConnector.Object);
+            _mockContainer.SetupGet(c => c.Responder).Returns(_mockResponder.Object);
+            _mockResponder.SetupGet(r => r.SuperRoot).Returns(_superRootNode);
+            _mockResponder.SetupGet(r => r.SubscriptionManager).Returns(_mockSubManager.Object);
+
+            _mockConnector.Setup(c => c.AddValueUpdateResponse(It.IsAny<JToken>()))
+                .Returns(Task.FromResult(false));
+        }
+
+        [Test]
+        public void TestBannedCharacters()
+        {
+            var bannedChars = Node.BannedChars;
+            foreach (char c in bannedChars)
+            {
+                Assert.Throws<ArgumentException>(() =>
+                {
+                    _superRootNode.CreateChild($"Test{c}").BuildNode();
+                });
+            }
+
+            var multiCharTest = new string(Node.BannedChars);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                _superRootNode.CreateChild(multiCharTest).BuildNode();
+            });
+
+            var noCharTest = "TestNoBannedChars";
+            Assert.DoesNotThrow(() =>
+            {
+                _superRootNode.CreateChild(noCharTest).BuildNode();
+            });
+        }
+
+        [Test]
+        public void TestCreateChildren()
+        {
+            int first = 5; // Number of children of root to create
```

Comment:
Why not call those variables already `AmountOfFirstLevelNodes` and `AmountOfSecondLevelNodes` or something like that. I feel like the comment pretty much shows that the name is unclear.